### DISCUSSION TITLE
When using different domains, is it possible to only generate pages for one locale?

### DISCUSSION THREAD

**Discussion by Guito** [nuxt-modules/i18n#2304] [Q&A]
I am using i18n different domains for a long time now (https://github.com/nuxt-modules/i18n/issues/19). I am migrating my application to nuxt 3 and I was wondering if it's possible to generate only the routes related to one locale but also have the urls for the rest of locales. So, for example:

Locales:
```
const locales = [
    {
        code: 'en',
        name: 'English',
        langFile: 'en.json',
        domain: en.localhost
    },
    {
        code: 'es',
        name: 'Español',
        langFile: 'es.json',
        domain: es.localhost
    }
]
```

Page example:
```
const {locale, locales, t} = useI18n()
const links = locales.value.map(locale => ({
    hid: 'alternate-hreflang-' + locale.code,
    rel: 'alternate',
    href: switchLocalePath(locale.code),
    hreflang: locale.code,
}));
links.push({
    rel: 'canonical', href: switchLocalePath(locale.value), id: 'canonical'
});
useHead({
    link: links
})

defineI18nRoute({
    paths: {
        en: '/page',
        es: '/pagina,
    }
})
```

nuxt config adding a new param called **generateOnlyDefaultLocale**
```
i18n: {
        strategy: 'prefix',
        defaultLocale: 'en',
        locales: locales,
        differentDomains: true,
        loadLanguagesAsync: true,
        langDir: 'assets/locales/',
        lazy: true,
        generateOnlyDefaultLocale: true
    }
```

I would expect, when running generate, to only generate the file 'page' and not 'pagina' and the generated file should have meta tags to 'pagina'

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution) (The contribution guideline of nuxt-modules/i18n is compliant with Nuxt too).
- [X] Check existing [discussions](https://github.com/nuxt-modules/i18n/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by kazupon** [comment]
This is a question, not issue.
We will move to github discussions.

**Reply by kazupon** [reply]
nuxt i18n module v8 doesn't have `generateOnlyDefaultLocale` option...

**Reply by Guito** [reply]
Yes, I know it does not have it. I meant that it could be an option to add to the module. Would it be possible to do something like this?
